pkg/controller/direct/spanner/instance_controller.go

```diff
@@ -0,0 +1,284 @@
+	return &modelSpannerInstance{config: *config}, nil
+}
+
+var _ directbase.Model = &modelSpannerInstance{}
+
+type modelSpannerInstance struct {
+	config config.ControllerConfig
+}
+
+func (m *modelSpannerInstance) client(ctx context.Context) (*gcp.InstanceAdminClient, error) {
+	var opts []option.ClientOption
+	opts, err := m.config.RESTClientOptions()
+	if err != nil {
+		return nil, err
+	}
+	gcpClient, err := gcp.NewInstanceAdminRESTClient(ctx, opts...)
+	if err != nil {
+		return nil, fmt.Errorf("building Instance client: %w", err)
+	}
+	return gcpClient, err
+}
+
+func (m *modelSpannerInstance) AdapterForObject(ctx context.Context, reader client.Reader, u *unstructured.Unstructured) (directbase.Adapter, error) {
+	obj := &krm.SpannerInstance{}
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(u.Object, &obj); err != nil {
+		return nil, fmt.Errorf("error converting to %T: %w", obj, err)
+	}
+
+	id, err := krm.NewSpannerInstanceRef(ctx, reader, obj, u)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get spanner GCP client
+	gcpClient, err := m.client(ctx)
+	if err != nil {
+		return nil, err
+	}
+	resourceID := direct.ValueOf(obj.Spec.ResourceID)
+	if resourceID == "" {
+		resourceID = obj.GetName()
+	}
+	if resourceID == "" {
+		return nil, fmt.Errorf("cannot resolve resource ID")
+	}
+	return &SpannerInstanceAdapter{
+		id:         id,
+		gcpClient:  gcpClient,
+		desired:    obj,
+		resourceID: resourceID,
+	}, nil
+}
+
+func (m *modelSpannerInstance) AdapterForURL(ctx context.Context, url string) (directbase.Adapter, error) {
+	// TODO: Support URLs
+	return nil, nil
+}
+
+type SpannerInstanceAdapter struct {
+	id         *krm.SpannerInstanceRef
+	gcpClient  *gcp.InstanceAdminClient
+	desired    *krm.SpannerInstance
+	actual     *spannerpb.Instance
+	resourceID string
+}
+
+var _ directbase.Adapter = &SpannerInstanceAdapter{}
+
+func (a *SpannerInstanceAdapter) Find(ctx context.Context) (bool, error) {
+	log := klog.FromContext(ctx)
+	log.V(2).Info("getting SpannerInstance", "name", a.id.External)
+
+	req := &spannerpb.GetInstanceRequest{Name: a.id.External}
+	instancepb, err := a.gcpClient.GetInstance(ctx, req)
+	if err != nil {
+		if direct.IsNotFound(err) {
+			return false, nil
+		}
+		return false, fmt.Errorf("getting SpannerInstance %q: %w", a.id.External, err)
+	}
+
+	a.actual = instancepb
+	return true, nil
+}
+
+func (a *SpannerInstanceAdapter) Create(ctx context.Context, createOp *directbase.CreateOperation) error {
+	log := klog.FromContext(ctx)
+	log.V(2).Info("creating Instance", "name", a.id.External)
+	mapCtx := &direct.MapContext{}
+
+	desired := a.desired.DeepCopy()
+	if err := a.SpecValidation(); err != nil {
+		return err
+	}
+	resource := SpannerInstanceSpec_ToProto(mapCtx, &desired.Spec, a.id.SpannerInstanceConfigPrefix())
+	// If node count or processing unit is not specify,
+	// Default NodeCount to 1.
+	if resource.NodeCount == 0 && resource.ProcessingUnits == 0 {
+		resource.NodeCount = 1
+	}
```

Comment:
IIRC, the GCP service sets the default nodeCount=1, processingUnites 1000. If so, KCC does not need to duplicate this default.